### PR TITLE
Matched a couple of effects functions

### PR DESCRIPTION
**Pull request by mrshigure**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀 This PR matches 6 functions (+0.12%) on `us`.

File: src/effects/attack_result_text.c [hunk 1]
```diff
@@ -171,35 +171,16 @@ void func_E0090428(EffectInstance* effect) {
...
-    // u32 phi_v1_2;
-    // u32 phi_v0;
-    // u32 phi_v0_2;
-    // s32 phi_a0_2;
-    // s32 phi_v1_3;
-    // s32 phi_v0_3;
-
-    data = effect->data;
+
+    data = effect->data.attackResultText;
```
Comment (ethteck) [inline src/effects/attack_result_text.c:183]: Could you move the data temp to the top of the list and initialize it on that same line? same with unk_00 and unk_1C following. sp18 and sp58 should be okay after those 3

File: src/effects/attack_result_text.c [hunk 2]
```diff
@@ -171,35 +171,16 @@ void func_E0090428(EffectInstance* effect) {
     func_E0090444(effect);
 }
 
-// gSPTextureRectangle needs a lot of work
-#ifdef NON_EQUIVALENT
 void func_E0090444(EffectInstance* effect) {
```
Comment (ethteck) [inline src/effects/attack_result_text.c:174]: great job matching this!!

File: src/effects/chapter_change.c [hunk 3]
```diff
@@ -1,15 +1,60 @@
...
+void shim_draw_msg(s32, s32, s32, s32, s32, s32);
 s32 shim_get_msg_width(s32, u16);
 
-INCLUDE_ASM(s32, "effects/chapter_change", func_E010E000);
+void func_E010E000(ChapterChangeFXData* data, s32 arg1, UnkStruct* arg2) {
+    UnkStruct* it;
+    f32 var0;
+    f32 var1;
+    s32 temp;
+    s32 unk_1C = data->unk_1C;
```
Comment (ethteck) [inline src/effects/chapter_change.c:31]: could you move unk_1C temp to the top?

File: src/effects/confetti.c [hunk 4]
```diff
@@ -3,14 +3,77 @@
...
+}
 
 INCLUDE_ASM(s32, "effects/confetti", confetti_main);
 
 void confetti_init(void) {
 }
 
-INCLUDE_ASM(s32, "effects/confetti", confetti_update);
+void confetti_update(EffectInstance* effect) {
+    s32 i;
```
Comment (ethteck) [inline src/effects/confetti.c:22]: part temp should come first and i temp should be last

**Review by ethteck — CHANGES_REQUESTED** [review]
Fantastic job with these!

**Comment by mrshigure** [comment]
Comments addressed!

**Review by ethteck — APPROVED** [review]

**Review by nanaian — APPROVED** [review]

**Pull request merged**